index.js

```diff
@@ -486,4 +483,31 @@ FeatureService.prototype._catchErrors = function (task, e, url, cb) {
   }.bind(this), task.retry * 1000)
 }
 
+/* Forms a standardized error object for reporting back up the chain
+ * @param {object} error - http request error
+ * @param {string} message - human readable message about ther
+ * @param {object} json - json returned from server request
+ * @returns {object} - error object containing diagnostic information
+ */
+FeatureService.prototype._buildError = function (url, message, error) {
+  if (!error) error = {}
+  error.message = message
+  error.request = url
+  error.response = error.message
+  error.type = error.type || 'Unknown failure'
+  error.code = error.code || 500
+
+  return error
+}
```

Comment:
So I have some issues with this implementation.

First and foremost, why not use the [`Error`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) that is already in JavaScript? Errors should be Errors, not arbitrary strings or arbitrary objects. If our errors really need type, request, and response properties, we can create a new Error prototype that inherits from the base Error constructor.

Second, why create a property called "response" that is literally a copy of "message"? Why not use message?

Third, "type" looks a lot like a message to me, not a type. Something like https://github.com/koopjs/featureservice/pull/12/files#diff-168726dbe96b3ce427e7fedce31bb0bcR439 doesn't have a message and appears to pass in a type that would be a lot more useful as a message.

Can I take a crack at showing how to make an Error constructor that extends the base js Error prototype?